src/analysis/taint_analysis.rs

```diff
@@ -210,28 +245,72 @@ where
             Some(AttrInfoKind::Source) => self.t_visit_source_destination(destination),
             Some(AttrInfoKind::Sanitizer) => self.t_visit_sanitizer_destination(destination),
             Some(AttrInfoKind::Sink) => self.t_visit_sink(name, args, span),
-            None => {
-                // TODO(Hilmar): Perform analysis
+            None => self.t_visit_analysis(args, id, destination),
+        }
+    }
+
+    fn t_visit_analysis(
+        &mut self,
+        args: &[Operand],
+        id: &rustc_hir::def_id::DefId,
+        destination: &Option<(Place, BasicBlock)>,
+    ) {
+        let init = args
+            .iter()
+            .map(|arg| match arg {
+                Operand::Copy(p) | Operand::Move(p) => Some(self.state.get_taint(p.local)),
+                Operand::Constant(_) => None,
+            })
+            .collect::<Vec<_>>();
+        let target_body = self.tcx.optimized_mir(*id);
+        let mut results = TaintAnalysis::new_with_init(self.tcx, self.info, init)
+            .into_engine(self.tcx, target_body)
+            .pass_name("taint_analysis")
+            .iterate_to_fixpoint()
+            .into_results_cursor(target_body);
+        if let Some(last) = target_body.basic_blocks().last() {
+            results.seek_to_block_end(last);
+            let end_state = results.get();
+
+            // Check if return place is tainted, in which case this is a typical sink.
+            if end_state.get_taint(Local::from_usize(0)) {
+                self.t_visit_source_destination(destination);
+            }
```

Comment:
Do you mean that it is a source, or am i misunderstanding something?

Reply:
ah yes, I keep mixing them up